### DISCUSSION TITLE
Assistance Required for PyFluent Workflow Issues

### DISCUSSION THREAD

**Discussion by Paul090196** [ansys/pyfluent#1890] [Q&A]
![interface zone type](https://github.com/ansys/pyfluent/assets/138394566/9bcbcbc1-7587-4907-9026-e77929bba8a5)
![zone_type](https://github.com/ansys/pyfluent/assets/138394566/5fa9b88a-34c0-4d43-beed-c6306950a7e4)
![interface zone create](https://github.com/ansys/pyfluent/assets/138394566/aa0f707c-4a98-4f3e-88cb-fffc44f97b64)
![1](https://github.com/ansys/pyfluent/assets/138394566/63beb441-aa44-4491-a5a6-a0673829352c)
![2](https://github.com/ansys/pyfluent/assets/138394566/3909d200-59e3-4f32-b318-886ad9fa95e0)
Dear Sir/Madam,

I hope this message finds you well. I am writing to seek your assistance with some issues I have encountered while working on the automation of my workflow using PyFluent for my simulation project. Your expertise would be highly appreciated in resolving these challenges.

The primary focus of my simulation involves utilizing frame motion and creating boundary conditions as interfaces. While constructing the mesh interfaces, I am trying to implement the "Periodic boundary condition" feature under the interface options. However, I couldn't locate the appropriate keyword or option for selecting this condition. I have attached an image that illustrates the desired outcome. Your guidance in this matter would be immensely valuable.

Additionally, I am facing difficulty in accessing the "compute from" option, which is available under the reference value and initialization tab. This option is essential for my simulation, as I need to select the names assigned to the boundaries. Without the "compute from" option, I am unable to initialize my simulation with the assigned values. Your insight on how to access this option and effectively use it would be of great assistance.

Lastly, I would greatly appreciate your guidance on how to assign the "print to the console" and "report file" options under the report definition of Lift and Drag coefficients. These outputs are crucial for my analysis, and having a clear understanding of how to set up these options within the PyFluent framework would greatly enhance the accuracy of my results.

To facilitate a better understanding of the challenges I am facing, I have attached relevant images to this email. Your expertise in addressing these issues would significantly contribute to the successful completion of my simulation project.

Thank you for considering my request for assistance. I value your input and guidance, and I am looking forward to your response at your earliest convenience.

Best regards,
PaulStephen

**Comment by mkundu1** [comment]
If you're using Fluent 23.2, a settings API for mesh interfaces is available at `<solver_session>.setup.mesh_interfaces`. You can use the TUI `<solver_session>.tui.setup.mesh_interfaces` in previous Fluent versions.

The "compute from" command is available in the path `<solver_session>.setup.reference_values.compute`.

The `report_files` and `report_plots` are available under `<solver_session>.solution.monitor`. You can assign report definitions and set the print option within those objects.

There is a global search function now available in the develop branch. If you're building pyfluent from the latest develop branch (see the development installation section of https://github.com/ansys/pyfluent#installation), you can use it to search for a command/setting within API:
```
>>> import ansys.fluent.core as pyfluent
>>> pyfluent.search("mesh_interface") 
<solver_session>.tui.define.models.radiation.s2s_parameters.enable_mesh_interface_clustering (Command)
<solver_session>.tui.define.mesh_interfaces (Object)
<solver_session>.tui.define.overset_interfaces.options.mesh_interfaces (Command)
<solver_session>.setup.models.radiation.s2s.clustering_settings.enable_mesh_interface_clustering (Parameter)
<solver_session>.setup.mesh_interfaces (Object)
```
Note that, to use this feature, you must generate the API files by executing `python codegen/allapigen.py` as mentioned in the development installation.
